addons/nuxeo-platform-video/nuxeo-platform-video-core/src/test/java/org/nuxeo/ecm/platform/video/extension/TestVideoImporterAndListeners.java

```diff
@@ -386,14 +388,23 @@ public void testVideoUpdate() throws Exception {
 
     @Test
     @Deploy("org.nuxeo.ecm.platform.video.core:test-video-conversions-enabled.xml")
+    @Deploy("org.nuxeo.ecm.platform.video.core:test-video-listeners-contrib.xml")
     @SuppressWarnings("unchecked")
     public void testVideoConversions() throws IOException, InterruptedException {
         DocumentModel doc = session.createDocumentModel("/", "testVideoDoc", VIDEO_TYPE);
         Blob blob = getBlobFromPath("test-data/sample.mpg", "video/mpeg");
         doc.setPropertyValue("file:content", (Serializable) blob);
         doc = session.createDocument(doc);
 
+        // Init the KV and the counter used by the conversion done listener
+        KeyValueStore kv = Framework.getService(KeyValueService.class).getKeyValueStore("default");
+        kv.put(doc.getId(), 0L);
```

Comment:
You should have a look at the test listener first, the code is trivial it maintains a counter per docId which is incremented when receiving the event videoConversionDone.
The test makes sure the counter is set to 0 at the beginning.
Your confusion with the GroupJoin logic is because you use a debugger to understand the test logic which proves that the test is not readable in the first place.
What do you suggest?

Reply:
yep, perhaps complete with `see  #VideoConversionDoneListener ` :)